s2/builder_graph_options.go

```diff
@@ -0,0 +1,179 @@
+	// collection of adjacent polygons (a polygon mesh).
+	siblingPairsRequire
+	// siblingPairsCreate ensures that all edges have a sibling edge by
+	// creating them if necessary. This is useful with polygon meshes where
+	// the input polygons do not cover the entire sphere. Such edges always
+	// have an empty set of labels and do not have an associated InputEdgeID.
+	siblingPairsCreate
+)
+
+// graphOptions is only needed by Layer implementations. A layer is
+// responsible for assembling an Graph of snapped edges into the
+// desired output format (e.g., an Polygon). The graphOptions allows
+// each Layer type to specify requirements on its input graph: for example, if
+// degenerateEdgesDiscard is specified, then Builder will ensure that all
+// degenerate edges are removed before passing the graph to Layer's Build
+// method.
+type graphOptions struct {
+	// Specifies whether the Builder input edges should be treated as
+	// undirected. If true, then all input edges are duplicated into pairs
+	// consisting of an edge and a sibling (reverse) edge. Note that the
+	// automatically created sibling edge has an empty set of labels and does
+	// not have an associated InputEdgeId.
+	//
+	// The layer implementation is responsible for ensuring that exactly one
+	// edge from each pair is used in the output, i.e. *only half* of the graph
+	// edges will be used. (Note that some values of the siblingPairs option
+	// automatically take care of this issue by removing half of the edges and
+	// changing edgeType to Directed.)
+	//
+	// DEFAULT: edgeTypeDirected
+	edgeType edgeType
+
+	// DEFAULT: degenerateEdgesKeep
+	degenerateEdges degenerateEdges
+
+	// DEFAULT: duplicateEdgesKeep
+	duplicateEdges duplicateEdges
+
+	// DEFAULT: siblingPairsKeep
+	siblingPairs siblingPairs
+
+	// This is a specialized option that is only needed by clients that want to
+	// work with the graphs for multiple layers at the same time (e.g., in order
+	// to check whether the same edge is present in two different graphs). [Note
+	// that if you need to do this, usually it is easier just to build a single
+	// graph with suitable edge labels.]
+	//
+	// When there are a large number of layers, then by default Builder builds
+	// a minimal subgraph for each layer containing only the vertices needed by
+	// the edges in that layer. This ensures that layer types that iterate over
+	// the vertices run in time proportional to the size of that layer rather
+	// than the size of all layers combined. (For example, if there are a
+	// million layers with one edge each, then each layer would be passed a
+	// graph with 2 vertices rather than 2 million vertices.)
+	//
+	// If this option is set to false, this optimization is disabled. Instead
+	// the graph passed to this layer will contain the full set of vertices.
+	// (This is not recommended when the number of layers could be large.)
+	//
+	// DEFAULT: true
```

Comment:
Could we change the name so the default can be false, maybe `disableVertexFiltering`?